administrator/components/com_scheduler/src/Table/TaskTable.php

```diff
@@ -107,16 +105,21 @@ public function check(): bool
 	}
 
 	/**
-	 * @param   boolean  $updateNulls  True to update fields even if they're null [?]
+	 * Override parent store method.
+	 * Defaults to updating null fields.
+	 * This is needed because some datetime fields might need to be updated to null but AdminModel::save() does
+	 * not expose an option to pass true to Table::store()
+	 *
+	 * @param   boolean  $updateNulls  True to update fields even if they're null
 	 *
 	 * @return boolean  True if successful [yes?]
 	 *
 	 * @since  __DEPLOY_VERSION__
-	 * @throws Exception
+	 * @throws \Exception
 	 */
-	public function store($updateNulls = false): bool
+	public function store($updateNulls = true): bool
 	{
-		$date = Factory::getDate()->toSql();
+		$date   = Factory::getDate()->toSql();
 		$userId = Factory::getApplication()->getIdentity();
```

Comment:
You are using User object for `$userId` instead of ID.

Isn't it simpler?

```
	public function store($updateNulls = true): bool
	{
		// Set creation date if not set
		if (!(int) $this->created)
		{
			$this->created = Factory::getDate()->toSql();
		}

		// @todo : Should we add modified, modified_by fields? [ ]

		// Set created_by if needed
		if (empty($this->created_by))
		{
			$this->created_by = Factory::getApplication()->getIdentity()->id;
		}

		return parent::store($updateNulls);
	}
```

Reply:
Absolutely, and thanks for the bug spot!